File: src/test/java/net/haesleinhuepf/clij/ops/examples/Convert.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_convertFloat.CLIJ_convertFloat;
import net.haesleinhuepf.clij.ops.CLIJ_convertUInt16.CLIJ_convertUInt16;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.imagej.ImageJ;

public class Convert {

	@Test
	public void run() throws IOException {
		ImageJ ij = new ImageJ();
		Object blobs = ij.io().open("https://bds.mpi-cbg.de/samples/blobs.png");
		ij.ui().show("input", blobs);
		// push images to GPU
		Object blobsGPU = ij.op().run(CLIJ_push.class, blobs);

		Object blobsFloat = ij.op().run(CLIJ_convertFloat.class, blobsGPU);
		Object blobsUInt8 = ij.op().run(CLIJ_convertUInt16.class, blobsFloat);
		Object blobsUInt16 = ij.op().run(CLIJ_convertUInt16.class, blobsUInt8);

		// show result
		Object result = ij.op().run(CLIJ_pull.class, blobsUInt16);
		ij.ui().show("UInt16", result);

		// cleanup
		ij.op().run(CLIJ_close.class, blobsGPU);
		ij.op().run(CLIJ_close.class, blobsFloat);
		ij.op().run(CLIJ_close.class, blobsUInt8);
		ij.op().run(CLIJ_close.class, blobsUInt16);
		ij.op().run(CLIJ_close.class);

	}

	public static void main(String... args) throws IOException {
		Convert task = new Convert();
		task.run();
	}
}

File: src/test/resources/jython/convert.py
# CLIJ example ImageJ Ops jython: convert.py
#
# This script shows how to convert an image in the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops

blobs = io.open("https://bds.mpi-cbg.de/samples/blobs.png")
ui.show("input", blobs)
# push images to GPU
blobsGPU = ops.run("CLIJ_push", blobs)

blobsFloat = ops.run("CLIJ_convertFloat", blobsGPU)
blobsUInt8 = ops.run("CLIJ_convertUInt16", blobsFloat)
blobsUInt16 = ops.run("CLIJ_convertUInt16", blobsUInt8)

# show result
result = ops.run("CLIJ_pull", blobsUInt16)
ui.show("UInt16", result)

# cleanup
ops.run("CLIJ_close", blobsGPU)
ops.run("CLIJ_close", blobsFloat)
ops.run("CLIJ_close", blobsUInt8)
ops.run("CLIJ_close", blobsUInt16)

